### PR TITLE
Support finding RET instructions on ARM

### PR DESCRIPTION
**Pull request by JamesMBartlett**
Summary: Adds support for finding ret instructions for aarch64.
Also, uses the ELF file's machine type to determine the disassembly context triple (and which ret instructions to look for), instead of basing it off of which architecture ELFReader was built for.

Relevant Issues: #147

Type of change: /kind cleanup

Test Plan: Manually checked that it can properly find the ret instruction for the prebuilt_test_exe. Need to update the elf_reader_test to actually test based on an aarch64 ELF file instead of just the x86_64 prebuilt_test_exe.

### CODE REVIEW

File: src/stirling/obj_tools/elf_reader.cc [hunk 1]
```diff
@@ -519,12 +570,26 @@ std::vector<uint64_t> FindRetInsts(utils::u8string_view byte_code) {
   return res;
 }
 
+StatusOr<Arch> GetArchFromELFMachine(ELFIO::Elf_Half machine) {
+  switch (machine) {
+    case ELFIO::EM_X86_64:
+      return ArchX86_64;
+    // ELFIO doesn't have the updated versions of EM_*, so we use the aarch64 value directly.
+    case 183:
```
Comment (vihangm) [inline src/stirling/obj_tools/elf_reader.cc:578]: should we either update our fork with main or just update this one line in our fork from main?
https://github.com/serge1/ELFIO/blob/main/elfio/elf_types.hpp#L250

**Review by vihangm — APPROVED** [review]

**Comment by ddelnano** [comment]
I agree that we should have an updated elf_reader_test to verify this works with an aarch64 elf file.

In addition to that have you ran the BPF tests on arm? My understanding is that this return instruction code is important for Go uretprobes, so I'd like to see that our go uprobe tests are pass for both architectures.

**Comment by JamesMBartlett** [comment]
> I agree that we should have an updated elf_reader_test to verify this works with an aarch64 elf file.
> 
> 
> 
> In addition to that have you ran the BPF tests on arm? My understanding is that this return instruction code is important for Go uretprobes, so I'd like to see that our go uprobe tests are pass for both architectures.

The BPF tests don't currently work on ARM since the docker images we use for them are all built for x86. I was planning on delaying getting those to work until @zasgar qemu stuff lands, so that we can use an x86 machine to run them. I will manually verify that a go retprobe works.

**Comment by JamesMBartlett** [comment]
I tested that go TLS probes work on ARM (at least for HTTP2 traffic). There seems to be some issue on both ARM and x86 with go TLS probes for HTTP1. But dom is looking into that separately.

**Pull request merged**